pkg/executor/infoschema_reader.go

```diff
@@ -242,11 +242,29 @@ func getAutoIncrementID(
 	sctx sessionctx.Context,
 	tblInfo *model.TableInfo,
 ) int64 {
+	if raw, ok := is.(*infoschema.SessionExtendedInfoSchema); ok {
```

Comment:
Why do we need this type assertion?

Reply:
`Infoschema` that we can get from session is wrapped with `SessionExtendedInfoSchema`:
 https://github.com/pingcap/tidb/blob/a5832b4689acf925173227ccd59d06a5acd75e5c/pkg/table/temptable/infoschema.go#L23